src/languageservice/services/yamlCompletion.ts

```diff
@@ -1673,7 +1673,8 @@ const isNumberExp = /^\d+$/;
 function convertToStringValue(param: unknown): string {
   let value: string;
   if (typeof param === 'string') {
-    value = param;
+    //support YAML spec 1.1 boolean values
+    value = ['on', 'off', 'true', 'false', 'yes', 'no'].includes(param.toLowerCase()) ? `"${param}"` : param;
```

Comment:
It seems there is a configuration `yaml.format.singleQuote` that dictates if `"` or `'`. I think that should be respected.